From 3d4a4a3d62815f90fc65a827a3e2de96c4571350 Mon Sep 17 00:00:00 2001
From: Orlando Chamberlain <orlandoch.dev@gmail.com>
Date: Mon, 20 Nov 2023 10:32:23 +1100
Subject: [PATCH 1/1] acpi video: force native for some T2 macbooks

The intel backlight is needed for these.

MacBookPro15,2/4 or MacBookPro16,3 or MacBookAir8,1/2 might also need
this so I'm not going to be submitting this upstream  yet

mbp16,3 was reported not to have this issue for 6.5.8 at least.
---
 drivers/acpi/video_detect.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

Index: linux-lernstick/drivers/acpi/video_detect.c
===================================================================
--- linux-lernstick.orig/drivers/acpi/video_detect.c	2024-05-01 13:19:15.001803711 +0200
+++ linux-lernstick/drivers/acpi/video_detect.c	2024-05-01 13:19:15.001803711 +0200
@@ -540,6 +540,14 @@
 		},
 	},
 	{
+	 .callback = video_detect_force_native,
+	 /* Apple MacBook Air 9,1 */
+	 .matches = {
+		DMI_MATCH(DMI_SYS_VENDOR, "Apple Inc."),
+		DMI_MATCH(DMI_PRODUCT_NAME, "MacBookAir9,1"),
+		},
+	},
+	{
 	 /* https://bugzilla.redhat.com/show_bug.cgi?id=1217249 */
 	 .callback = video_detect_force_native,
 	 /* Apple MacBook Pro 12,1 */
@@ -549,6 +557,14 @@
 		},
 	},
 	{
+	 .callback = video_detect_force_native,
+	 /* Apple MacBook Pro 16,2 */
+	 .matches = {
+		DMI_MATCH(DMI_SYS_VENDOR, "Apple Inc."),
+		DMI_MATCH(DMI_PRODUCT_NAME, "MacBookPro16,2"),
+		},
+	},
+	{
 	 .callback = video_detect_force_native,
 	 /* Dell Inspiron N4010 */
 	 .matches = {
